src/components/AddressSummary.js

```diff
@@ -0,0 +1,220 @@
+    const queryParams = this.props.pagination.obtainQueryParams();
+    queryParams.token = token;
+    const newURL = this.props.pagination.paginationUrl(queryParams);
+    this.props.pushNewURL(newURL);
+  }
+
+  updateNumberOfTransactions = (numberOfTransactions) => {
+    this.setState({ numberOfTransactions });
+  }
+
+  render() {
+    if (Object.keys(this.state.balance).length === 0) {
+      if (this.state.loading) {
+        return null;
+      }
+    }
+
+    const loadMainInfo = () => {
+      return (
+        <div className="card text-white bg-dark mb-3">
+          <div className="card-body">
+            Address: {this.props.address}<br />
+            Number of tokens: {Object.keys(this.state.balance).length}
+          </div>
+        </div>
+      );
+    }
+
+    const loadBalanceInfo = () => {
+      const balance = this.state.balance[this.state.selectedToken];
+      return (
+        <div className="card bg-light mb-3">
+          <div className="card-body">
+            Token: {renderTokenData()}<br />
+            Number of transactions: {this.state.numberOfTransactions}<br />
+            Total received: {hathorLib.helpers.prettyValue(balance.received)}<br />
+            Total spent: {hathorLib.helpers.prettyValue(balance.spent)}<br />
+            <strong>Final balance: </strong>{hathorLib.helpers.prettyValue(balance.received - balance.spent)}
+          </div>
+        </div>
+      );
+    }
+
+    const renderTokenData = () => {
+      if (Object.keys(this.state.balance).length === 1) {
+        const balance = this.state.balance[this.state.selectedToken];
+        return <span>{balance.name} ({balance.symbol})</span>
+      } else {
+        return (
+          <select value={this.state.selectedToken} onChange={this.selectChanged}>
+            {renderTokenOptions()}
+          </select>
+        );
+      }
+    }
+
+    const renderTokenOptions = () => {
+      return Object.keys(this.state.balance).map((uid) => {
+        const tokenData = this.state.balance[uid];
+        return <option value={uid} key={uid}>{tokenData.name} ({tokenData.symbol})</option>;
+      });
+    }
+
+    const loadSummary = () => {
+      return (
+        <div>
+          {loadMainInfo()}
+          {loadBalanceInfo()}
+        </div>
+      );
+    }
+
+    return (
+      <div className="w-100">
+        {this.state.loading ? <ReactLoading type='spin' color={colors.purpleHathor} delay={500} /> : loadSummary()}
+        <p className="text-danger mt-3">{this.state.errorMessage}</p>
+      </div>
+    );
+  }
+}
+
+/*
+ * address: Address to show summary
+ * shouldUpdate: Function that receives a tx data and returns if summary should be updated
+ * pagination: Instance of pagination class that handles the URL parameters
+ */
+AddressSummary.propTypes = {
+  address: PropTypes.string.isRequired,
+  shouldUpdate: PropTypes.func.isRequired,
+  pagination: PropTypes.instanceOf(PaginationURL).isRequired,
+};
+
```

Comment:
I'll comment here so it's easier to follow a thread discussion.

> At the beginning this was not a big problem because it was a simple screen, even though I had to do some repeated checks in both components, it was not a big deal. After we decided to add a token filter, the code became more complex and I believe some parts of the code became a little bit confusing because I had to exchange information between siblings components. I thought about doing a refactor to put everything in a single component, however the code was already advanced and I wanted a first feedback from you.

Actually, having 2 separate components make sense. The problem I see is having the logic in the components. It'd be better form them to be functional components that only display the data and leave the logic to the screen element, that makes all API requests and updates components' props. 

So the screen element requests the balance and history and also passes functions as props to handle changes in the components, like you already have with `onRowClicked` (and others). I'd just follow this pattern until we remove all logic from components. I think this could greatly simplify the code.

Reply:
I agree that seeing the code now this might be the better approach but my idea was to make the two component 'plug and play'. If we needed to use them elsewhere we wouldn't need to execute the requests again, capture the ws message and everything else, we would just add the components there and everything would work. 

To do it, each component should be independent but I don't know if it makes sense to do it now, since it increases the complexity of the code. I think what I have right now is a mix of both approaches.

Reply:
It was not clear to me which approach you think has higher code complexity. I still believe that what I suggested is much simpler. Right now, we have AddressHistory updating an element in AddressSummary, for example.

I understand your point about it being self contained, but I think the logic should not be in this component.

Reply:
I agree with @obiyankenobi that the components should only display the results. All other things, including request management and business logic should be handled elsewhere.

It seems to require a refactor of many components/screens because we've been mixing these things all over the code in this project. Even so, maybe we should "do it right" and separate those things in the new components/screens.

Reply:
I've updated the code and now the screen has all the logic.

Can you do the refactor again, please? @msbrogli @obiyankenobi

Reply:
Since you two approved after the refactor, I assumed all is good. So resolving this